src/Compilers/Core/Portable/Symbols/SymbolEqualityComparer.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Symbols;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Allows for the comparison of two <see cref="ISymbol"/> instances
+    /// </summary>
+    public sealed class SymbolEqualityComparer : IEqualityComparer<ISymbol>
+    {
+        /// <summary>
+        /// Compares two <see cref="ISymbol"/> instances based on the default comparison rules, equivalent to calling <see cref="IEquatable{ISymbol}.Equals(ISymbol)"/>
+        /// </summary>
+        public readonly static SymbolEqualityComparer Default = new SymbolEqualityComparer(TypeCompareKind.AllNullableIgnoreOptions);
+
+        /// <summary>
+        /// Compares  two <see cref="ISymbol"/> instances, considering their nullability
+        /// </summary>
+        public readonly static SymbolEqualityComparer IncludeNullability = new SymbolEqualityComparer(TypeCompareKind.ConsiderEverything2); //TODO: should this be explicitly *not* compare everything
```

Comment:
>//TODO: should this be explicitly *not* compare everything [](start = 140, length = 58)

Do we have a bug tracking this? What should we compare rather than everything?

Reply:
Loosly tracked by #31742, hence using ConsiderEverything2. My thinking is it should probably be something like Default + ~AllIgnoreNullableOptions. Then as the underlying enum changes, we'll change with it to stay current.

---
In reply to: [305545502](https://github.com/dotnet/roslyn/pull/37247#discussion_r305545502) [](ancestors = 305545502)